src/generator.rs

```diff
@@ -96,9 +97,50 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
         }
     }
 
+    let compressors: BTreeSet<&String> = devices
+        .iter()
+        .filter(|device| device.compression_algorithm.is_some())
+        .flat_map(|device| device.compression_algorithm.as_ref())
+        .collect::<BTreeSet<_>>();
+
+    if !compressors.is_empty() {
+        let known = get_known_compressors()?;
+
+        for comp in compressors.into_iter().filter(|&c| !known.contains(c)) {
+            let status = Command::new("modprobe")
+                .arg(format!("crypto-{}", comp))
+                .status()
+                .context("Failed to spawn modprobe")?;
+            if !status.success() {
+                warn!("modprobe crypto-{} failed, ignoring: code {}", comp, status);
+            }
+        }
+    }
+
     Ok(())
 }
 
+fn _get_known_compressors(path: &Path) -> Result<BTreeSet<String>> {
+    let content = fs::read_to_string(path).with_context(|| format!("Failed to read {:?}", path))?;
+
+    // Modules with type of `compression` have a `-generic` suffix in the driver name
+    let available = content
+        .lines()
+        .into_iter()
+        .filter(|line| line.starts_with("name"))
+        .map(|m| m.rsplit(':').next().unwrap().trim())
+        .map(String::from)
+        .collect();
+
+    Ok(available)
+}
+
+// Returns a list of names of loaded compressors
```

Comment:
```suggestion
// Returns a list of names that includes loaded modules of compressors
```

Reply:
Actually, those are the names of *compressors*, not *module* names. (A module may implement multiple algorithms.) And we don't particularly care about modules themselves.